internal/pkg/cli/app_show.go

```diff
@@ -0,0 +1,312 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"math"
+	"os"
+	"strconv"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/archer"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/describe"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/store"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/color"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/log"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+const (
+	applicationShowProjectNamePrompt     = "Which project's applications would you like to show?"
+	applicationShowProjectNameHelpPrompt = "A project groups all of your applications together."
+	applicationShowAppNamePrompt         = "Which application of %s would you like to show?"
+	applicationShowAppNameHelpPrompt     = "The detail of an application will be shown (e.g., endpoint URL, CPU, Memory)."
+)
+
+// ShowAppOpts contains the fields to collect for showing an application.
+type ShowAppOpts struct {
+	ShouldOutputJSON bool
+
+	proj *archer.Project
+	app  *archer.Application
+	envs []*archer.Environment
+	uris []*describe.WebAppURI
+
+	appName string
+
+	projectLister archer.ProjectLister
+	projectGetter archer.ProjectGetter
+	appLister     archer.ApplicationLister
+	appGetter     archer.ApplicationGetter
+	envLister     archer.EnvironmentLister
+	identifier    ResourceIdentifier
+
+	w io.Writer
+
+	*GlobalOpts
+}
+
+// BuildAppShowCmd builds the command for showing applications in a project.
+func BuildAppShowCmd() *cobra.Command {
+	opts := ShowAppOpts{
+		w:          os.Stdout,
+		GlobalOpts: NewGlobalOpts(),
+	}
+	cmd := &cobra.Command{
+		Use:   "show",
+		Short: "Shows all the applications in a project",
+		Example: `
+  Shows detail for the application my-app
+  /code $ ecs-preview app show -a my-app`,
+		PreRunE: runCmdE(func(cmd *cobra.Command, args []string) error {
+			ssmStore, err := store.New()
+			if err != nil {
+				return fmt.Errorf("connect to environment datastore: %w", err)
+			}
+			opts.projectLister = ssmStore
+			opts.projectGetter = ssmStore
+			opts.appLister = ssmStore
+			opts.appGetter = ssmStore
+			opts.envLister = ssmStore
+
+			return nil
+		}),
+		RunE: runCmdE(func(cmd *cobra.Command, args []string) error {
+			if err := opts.Ask(); err != nil {
+				return err
+			}
```

Comment:
Should we be validating the application name and project names that are passed in?